### PR TITLE
Fix source secret data source tests

### PR DESCRIPTION
**Pull request by Shubbhang351**
1. **What this PR does / why we need it**:

Fix source secret data source tests

Output for Acceptance tests.
Screenshot 2023-09-26 at 10 47 34 AM

### CODE REVIEW

File: internal/resources/sourcesecret/data_sourcesecret_test.go [hunk 1]
```diff
@@ -41,39 +41,6 @@ func TestAcceptanceForSourceSecretDataSource(t *testing.T) {
 				Config: testConfig.getTestSourceSecretDataSourceBasicConfigValue(commonscope.ClusterScope, spec.SSHKey),
 				Check:  testConfig.checkSourceSecretDataSourceAttributes(),
 			},
-			{
```
Comment (ishangupta-ds) [inline internal/resources/sourcesecret/data_sourcesecret_test.go:44]: Don't we need this test? @Shubbhang351

File: internal/resources/sourcesecret/data_sourcesecret_test.go [hunk 2]
```diff
@@ -41,39 +41,6 @@ func TestAcceptanceForSourceSecretDataSource(t *testing.T) {
...
-				Config: testConfig.getTestSourceSecretDataSourceBasicConfigValue(commonscope.ClusterGroupScope, spec.SSHKey, WithKnownhosts("somehostes")),
-				Check:  testConfig.checkSourceSecretDataSourceAttributes(),
-			},
-		},
-	},
-	)
-
-	t.Log("source secret data source acceptance test complete for SSH credential type")
-
-	// Test case for source secret data source with Username/Password credential type.
```
Comment (ishangupta-ds) [inline internal/resources/sourcesecret/data_sourcesecret_test.go:62]: Why are we removing the tests?

**Review by ishangupta-ds — CHANGES_REQUESTED** [review]
Are the removed test cases not needed?

**Comment by ramya-bangera** [comment]
@Shubbhang351 - Can you capture what was the issue and what is the fix here in description?

File: internal/resources/sourcesecret/data_sourcesecret_test.go (hunk 1 above)
Comment (Shubbhang351) [reply internal/resources/sourcesecret/data_sourcesecret_test.go:44]: second tests I've added last time just to test spec SSH Type, which can be checked through update call so I've merged the tests for both spec type.

File: internal/resources/sourcesecret/data_sourcesecret_test.go (hunk 2 above)
Comment (Shubbhang351) [reply internal/resources/sourcesecret/data_sourcesecret_test.go:62]: tests which I've removed they were covering one specific spec type which can be tests through update scenario that's why I've merged those tests in to one.

File: internal/resources/sourcesecret/data_sourcesecret_test.go (hunk 1 above)
Comment (ishangupta-ds) [reply internal/resources/sourcesecret/data_sourcesecret_test.go:44]: Sure.

**Review by ramya-bangera — APPROVED** [review]

**Pull request merged**